packages/core/src/libraries/social.ts

```
@@ -47,33 +48,39 @@ const encryptTokenResponse = (tokenResponse?: TokenResponse): EncryptedTokenSet
     return;
   }
 
-  const {
-    access_token,
-    id_token,
-    refresh_token,
-    scope,
-    token_type: tokenType,
-    expires_in,
-  } = tokenResponse;
+  try {
+    const {
+      access_token,
+      id_token,
+      refresh_token,
+      scope,
+      token_type: tokenType,
+      expires_in,
+    } = tokenResponse;
 
-  const requestedAt = Math.floor(Date.now() / 1000);
+    const requestedAt = Math.floor(Date.now() / 1000);
 
-  const expiresAt = expires_in && requestedAt + expires_in;
+    const expiresAt = expires_in && requestedAt + expires_in;
 
-  const encryptedTokenSet = encryptTokens({
-    access_token,
-    ...conditional(id_token && { id_token }),
-    ...conditional(refresh_token && { refresh_token }),
-  });
+    const encryptedTokenSet = encryptTokens({
+      access_token,
+      ...conditional(id_token && { id_token }),
+      ...conditional(refresh_token && { refresh_token }),
+    });
 
-  return {
-    encryptedTokenSetBase64: serializeEncryptedSecret(encryptedTokenSet),
-    metadata: {
-      scope,
-      tokenType,
-      expiresAt,
-    },
-  };
+    return {
+      encryptedTokenSetBase64: serializeEncryptedSecret(encryptedTokenSet),
+      metadata: {
+        scope,
+        tokenType,
+        expiresAt,
+      },
+    };
+  } catch (error: unknown) {
+    // Token encryption should not break the normal social authentication flow
+    // Return undefined to indicate no token response is available
+    void appInsights.trackException(error);
```

do we need to throw the original error?